interpreter/src/symbol_table/mod.rs

```diff
@@ -90,6 +90,36 @@ impl Table {
         self.0
             .remove_if(|x| !(vars.contains(x) || x.is_intrinsic()));
     }
+
+    #[must_use]
+    pub fn data(&self, var: &MVar) -> (bool, bool) {
+        self.0
+            .locate(&var.name)
+            .map(|x| x.data(&var.key))
+            .unwrap_or_default()
+    }
+
+    //Returns a string representation of Key in the given MVar.
+    #[must_use]
+    pub fn query(&self, var: &MVar, direction: Direction) -> String {
+        self.0
+            .locate(&var.name)
+            .and_then(|data| data.query(&var.key, direction))
+            .map(|key| {
+                let mut next_var = var.clone();
+                next_var.key = key.clone();
+                format!("{next_var}")
+            })
+            .unwrap_or_default()
+    }
+
+    #[must_use]
+    pub fn order(&self, var: &MVar, direction: Direction) -> Value {
```

Comment:
Add documentation comment, consider returning a sub_key rather then a value. (have the caller do the conversion)